### DISCUSSION TITLE
Plausible giving "Request Header Fields Too Large" error for version 2.1.2, 2.1.3 and 2.1.4

### DISCUSSION THREAD

**Discussion by jaideepgaikwad** [plausible/analytics#4689] [Self-Hosted Support] [answered]
**Issue:**
Getting error "**(Ch.Error) Request Header Fields Too Large**" for community-edition v2.1.2 v2.1.3 and v2.1.4.

**Using official Plausible Cloud hosting or self-hosting?** 
Self-hosting

**Describe the issue** 
The application isn't starting up and is stuck in a crash loop when community edition version v2.1.2 or v2.1.3 or v2.1.4 used with bitnami/clickhouse or clickhouse-server.

```
Loading plausible..
Starting dependencies..
Starting repos..
Plausible.Repo database already exists
Plausible.IngestRepo database already exists
Creation of Db successful!
Loading plausible..
Starting dependencies..
Starting repos..
** (Ch.Error) Request Header Fields Too Large
    (ecto_sql 3.12.0) lib/ecto/adapters/sql.ex:1078: Ecto.Adapters.SQL.raise_sql_call_error/1
    (ecto_sql 3.12.0) lib/ecto/migration/runner.ex:341: Ecto.Migration.Runner.log_and_execute_ddl/3
    (elixir 1.17.1) lib/enum.ex:1703: Enum."-map/2-lists^map/1-1-"/2
    (stdlib 6.0) timer.erl:590: :timer.tc/2
    (ecto_sql 3.12.0) lib/ecto/migration/runner.ex:25: Ecto.Migration.Runner.run/8
    (ecto_sql 3.12.0) lib/ecto/migrator.ex:365: Ecto.Migrator.attempt/8
    (ecto_sql 3.12.0) lib/ecto/migrator.ex:282: anonymous fn/5 in Ecto.Migrator.do_up/5
```

community-edition v2.1.1 does works fine with bitnami/clickhouse or clickhouse-server without any error. I have also tried deploying fresh instance with v2.1.4 but getting same error.

**Expected behavior** 
Plausible should start up

**Screenshots**

![image](https://github.com/user-attachments/assets/8a9ae5de-291d-4750-a64d-80a590f65a57)

**Environment**
```
- Plausible: community-edition: 2.1.2, 2.1.3 & 2.1.4
- Clickhouse: bitnami/clickhouse:24.4.1-debian-12-r4
- Clickhouse-server: clickhouse/clickhouse-server:24.9.2.42-alpine

**Comment by ruslandoga** [comment]
👋 @jaideepgaikwad 

Please describe your setup in more detail.

**Reply by ruslandoga** [reply, marked as answer]
The parameters we pass in the headers are [x-clickhouse-user, x-clickhouse-key, x-clickhouse-database, user-agent](https://github.com/plausible/ch/blob/ca21639a46d9b2c6a42a37eb2bf600eeaeea3e05/lib/ch/connection.ex#L370-L373), and [x-clickhouse-format](https://github.com/plausible/ch/blob/ca21639a46d9b2c6a42a37eb2bf600eeaeea3e05/lib/ch/query.ex#L132C31-L132C50) for SELECTs. It'd take extra effort to make them larger than what ClickHouse accepts. Maybe you have some sort of reverse-proxy between Plausible and ClickHouse?

**Reply by jaideepgaikwad** [reply]
@ruslandoga Thanks for quick reply and pointing to right direction. Istio-proxy was enabled in the namespace which was causing this issue, able to run v2.1.4 without any issue after disabling it.
